src/FFI-Kernel/ExternalAddress.class.st

```diff
@@ -86,7 +86,9 @@ ExternalAddress >> + offset [
 	"Convert xaddr -> bytes"
 	bytes := self asByteArrayPointer.
 	"Update bytes using platform dependent accessors"
-	bytes unsignedLongAt: 1 put: (bytes unsignedLongAt: 1) + offset.
+	self size = 4
+		ifTrue: [bytes unsignedLongAt: 1 put: (bytes unsignedLongAt: 1) + offset]
+		ifFalse: [bytes unsignedLongLongAt: 1 put: (bytes unsignedLongLongAt: 1) + offset].
```

Comment:
Why writing this by hand like that and not reusing `pointerAt:put:`?

Reply:
It's not clear to us that `size` will return the actual size of a word for an address...
Alternatively, there is also `platformLongAt:put:` that uses the platform size...